### PR TITLE
Refactor pull request CI

### PR DESCRIPTION
**Pull request by krzysz00**
Implements the compromise from https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/314

### CODE REVIEW

**Review by yiqian1 — APPROVED** [review]

File: mlir/utils/jenkins/Jenkinsfile.downstream [hunk 1]
```diff
@@ -1,65 +1,43 @@
...
+
 pipeline {
     agent {
         docker {
-            image 'rocm/mlir:rocm4.1-latest'
-            args '--user "$(id -u):$(id -g)" --device=/dev/kfd --device=/dev/dri --group-add video -u 0'
-            label 'rocm'
+            image dockerImage()
+            args dockerArgs()
+            label "rocm && !single_gpu"
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.downstream]: Why disable single gpu?

Comment (krzysz00) [reply mlir/utils/jenkins/Jenkinsfile.downstream]: We probably don't need to for this - I'll fix this later this afternoon.

File: mlir/utils/jenkins/Jenkinsfile [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// ON CHANGING THESE, ALSO CHANGE Jenkinsfile.downstream
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile:1]: Irrelevant to the code, are you planning on to remove the Jenkinsfile.downstream-xdlops in another PR?

File: mlir/utils/jenkins/Jenkinsfile.out-of-tree-dialect [hunk 3]
```diff
@@ -1,64 +0,0 @@
-pipeline {
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.out-of-tree-dialect:1]: Thanks for cleaning this up.

File: mlir/utils/jenkins/Jenkinsfile (hunk 2 above)
Comment (krzysz00) [reply mlir/utils/jenkins/Jenkinsfile:1]: Yeah, I'll detele `Jenkinsfile.dowstream-xdlops` once we've validated that this works.

File: mlir/utils/jenkins/Jenkinsfile.downstream (hunk 1 above)
Comment (krzysz00) [reply mlir/utils/jenkins/Jenkinsfile.downstream]: Done

**Review by jerryyin — APPROVED** [review]

**Pull request merged**